Source/CarthageKit/Xcode.swift

```diff
@@ -50,6 +50,16 @@ private func parseSwiftVersionCommand(output: String?) -> String? {
 	return "\(first) (\(second))"
 }
 
+/// Determines the Swift version of a framework at a given `URL`.
+internal func frameworkSwiftVersionIfIsSwiftFramework(_ frameworkURL: URL) -> SignalProducer<String?, SwiftVersionError> {
+	guard isSwiftFramework(frameworkURL) else {
+		return SignalProducer(value: nil)
+	}
+	return frameworkSwiftVersion(frameworkURL).flatMap(.merge, { (swiftVersion) -> SignalProducer<String?, SwiftVersionError> in
+		return SignalProducer(value: swiftVersion)
+	})
```

Comment:
This can be `frameworkSwiftVersion(frameworkURL).map(Optional.some)`

Reply:
Ammended my commit to simplify code : https://github.com/Carthage/Carthage/pull/2707/commits/602e99905e95284fde8d1f472bf6fbd622c1cfcf#diff-f66644e8cfbcad3c3826395a690ddfbcR58

Thanks